src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -673,18 +675,23 @@ public static boolean compareRow(List<Object> expectedRow, ResultSet actualRow)
                                             + (i + 1));
                             return false;
                         }
-                    } else {
+                    } else if (expected_obj instanceof BsonValue) {
                         Object actual_obj = actualRow.getObject(i + 1);
-                        if (!expected_obj.equals(actual_obj)) {
+                        MongoSQLValue expectedAsExtJsonValue =
+                                new MongoSQLValue((BsonValue) expected_obj);
+                        if (!expectedAsExtJsonValue.equals(actual_obj)) {
                             System.err.println(
-                                    "Expected Bson Other Object value "
+                                    "Expected Bson Other BsonValue value "
                                             + expected_obj
                                             + " but is "
                                             + actual_obj
                                             + " for column "
                                             + (i + 1));
                             return false;
                         }
+                    } else {
+                        throw new IllegalArgumentException(
+                                "unsupported expected value class: " + expected_obj.getClass());
```

Comment:
@bucaojit I left the previous case as `else if (expected_obj instanceof BsonValue)` so we can throw this more meaningful exception in case someone someday adds a test that has some other class as an expected result.